sw/device/silicon_creator/rom/rom_start.S

```diff
@@ -141,16 +144,72 @@ _rom_start_boot:
   wfi
   j .L_halt_loop
 
-.L_exec_en:
-  // Set up the global pointer.
-  //
-  // This requires that we disable linker relaxations, or it will be relaxed to
-  // `mv gp, gp`, so we disabled relaxations at the start of `_rom_start_boot`.
-  la gp, __global_pointer$
+  // Check if we are trying to transition to RMA. If so, busy loop until the
+  // transition is complete (up to CREATOR_SW_CFG_RMA_SPIN_CYCLES) and reset if
+  // the CPU is still executing.
+  li   a0, (TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET)
+  lw   t0, OTP_CTRL_PARAM_CREATOR_SW_CFG_RMA_SPIN_EN_OFFSET(a0)
+  li   t6, HARDENED_BOOL_TRUE
+  bne  t0, t6, .L_exec_en
 
-  // Re-enable linker relaxations.
-  .option pop
+  li   a1, (TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET)
+  lw   t1, OTP_CTRL_PARAM_CREATOR_SW_CFG_RMA_SPIN_EN_OFFSET(a1)
+  beq  t1, t6, .L_rma_spin_init
+  unimp
+  unimp
+  unimp
+  unimp
+
+.L_rma_spin_init:
```

Comment:
ah one thing, we need to make sure the ast init happens prior to this.
Otherwise, without accurate clocks, we won't be able to correctly transition `otp` or `flash`.
@zi-v do you have any disagreements here?

Reply:
No. I agree.

Reply:
SG, done. Just want to point out that ast init is also controlled by OTP.